src/oas.md

```diff
@@ -3447,7 +3447,7 @@ See examples for expected behavior.
 
 | Field Name | Type | Description |
 | ---- | :----: | ---- |
-| <a name="xml-name"></a>name | `string` | Replaces the name of the element/attribute used for the described schema property. When defined within `items`, it will affect the name of the individual XML elements within the list. When defined alongside `type` being `"array"` (outside the `items`), it will affect the wrapping element if and only if `wrapped` is `true`. If `wrapped` is `false`, it will be ignored. |
+| <a name="xml-name"></a>name | `string` | Replaces the name of the element/attribute used for the described schema property. For the root XML element, the name comes from the [schema component](#components-schemas) name; for other elements or attributes, the name comes from the property name.  When defined within `items`, it will affect the name of the individual XML elements within the list. When defined alongside `type` being `"array"` (outside the `items`), it will affect the wrapping element if and only if `wrapped` is `true`. If `wrapped` is `false`, it will be ignored. |
```

Comment:
Some implementations allow the `xml` fixed field in any Schema Object, including the Schema Object for the root XML element, and use its `name` value for the XML element name. If absent, the name of the XML element (or attribute) is the name of the name/value pair whose value is the Schema Object.

Which is straight-forward and what I would have expected, without the funny restriction to "property schemas", a term that is nowhere defined.

I'd not reinforce the surprising restriction to "property schemas" further, or at least add that implementations MAY allow it in any Schema Object, including the one for the XML root element.

Reply:
When we port this to 3.2 I was planning on removing the "property schemas" restriction, as I assumed the `xml` field in a "root schema" should behave exactly as you say.

I'm not sure how to incorporate that here in 3.1.2, though.  If @OAI/tsc is OK with relaxing that (as a MAY) in a patch release, I'm happy to do so.

For now, let me try to reword this short of adding that MAY (we can add the MAY in a later PR if we can't sort that out immediately- I'd love to see it, TBH).